components/coins-status/index.js

```diff
@@ -1,43 +1,88 @@
-import { useEffect } from 'react';
+import { useState, useEffect } from 'react';
 import { useAnimateValue } from './useCoinStatus';
 import styles from './coin.module.css';
 import PropTypes from 'prop-types';
 
+const BASE_API_URL = process.env.NEXT_PUBLIC_BASE_API_URL;
+const WALLET_URL = `${BASE_API_URL}/wallet`;
+const CURRENCIES = {
+  NEELAM: 'neelam',
+  DINERO: 'dinero',
+};
+
 const Coins = (props) => {
-  const [coins, changeCoins] = useAnimateValue(0, props.coin.value, 5000);
+  const { coin, balance } = props;
+  const [coins, changeCoins] = useAnimateValue(0, balance, 5000);
```

Comment:
Where do we use `coins`?

Reply:
coins used for animating balance. Now 'coins' renamed as 'animatedBalance', referred at right location. Also introduced new prop called 'animate' to provide component consumer to decide about animation.